From 6a27ec545622fd5dc7560e90b7069180fea9d331 Mon Sep 17 00:00:00 2001
From: "Lucie L. Hartmann" <lukas@mntre.com>
Date: Tue, 4 Nov 2025 00:48:23 +0100
Subject: [PATCH 05/11] reform-hw-setup: remove rk3588 ethernet hack

---
 bin/reform-hw-setup | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/bin/reform-hw-setup b/bin/reform-hw-setup
index 7f425a2..9e786da 100755
--- a/bin/reform-hw-setup
+++ b/bin/reform-hw-setup
@@ -54,19 +54,6 @@ init_a311d() {
   echo ffe03000.mmc >/sys/bus/platform/drivers/meson-gx-mmc/bind
 }
 
-init_rk3588() {
-  # shellcheck disable=SC2043
-  for m in dwmac_rk; do
-    grep --quiet "^$m " /proc/modules || continue
-    rmmod "$m"
-  done
-  # manual ethernet phy reset
-  gpioset -c 3 -t 100ms,100ms,0ms 23=1 \
-    || echo "E: gpioset for phy reset failed (requires gpiod >= 2.0)" >&2
-  sleep 0.5
-  modprobe dwmac_rk
-}
-
 init_wm8960() {
   # Fix WM8960 being too quiet
   amixer -c "$1" sset Playback 255
@@ -184,19 +171,15 @@ case "$(cat /proc/device-tree/model)" in
     init_qca9377_wifi || echo "W: init_qca9377_wifi failed" >&2
     ;;
   "MNT Reform 2 with RCORE RK3588 Module")
-    init_rk3588
     init_wm8960 rk3588wm8960
     ;;
   "MNT Reform 2 with RCORE-DSI RK3588 Module")
-    init_rk3588
     init_wm8960 rk3588wm8960
     ;;
   "MNT Pocket Reform with RCORE RK3588 Module")
-    init_rk3588
     init_tlv320aic3100 rk3588tlv320aic
     ;;
   "MNT Reform Next with RCORE RK3588 Module")
-    init_rk3588
     init_tlv320aic3100 rk3588tlv320aic
     ;;
   *)
-- 
2.47.3

