--- a/Makefile
+++ b/Makefile
@@ -134,10 +134,6 @@ install: $(MAN1) plymouth/background.png
 	$(INSTALLDATA) -t $(DESTDIR)$(datadir)/plymouth/themes/reform-y2k plymouth/background.png
 	$(INSTALLDATA) -t $(DESTDIR)$(datadir)/plymouth/themes/reform-y2k plymouth/reform-y2k.plymouth
 	$(INSTALL)     -d $(DESTDIR)$(libdir)/systemd/user
-	$(INSTALLDATA) -t $(DESTDIR)$(libdir)/systemd/user systemd/pipewire-pulse.service
-	$(INSTALLDATA) -t $(DESTDIR)$(libdir)/systemd/user systemd/pipewire-pulse.socket
-	$(INSTALLDATA) -t $(DESTDIR)$(libdir)/systemd/user systemd/pipewire.service
-	$(INSTALLDATA) -t $(DESTDIR)$(libdir)/systemd/user systemd/pipewire.socket
 	$(INSTALL)     -d $(DESTDIR)$(datadir)/wireplumber/wireplumber.conf.d
 	$(INSTALLDATA) -t $(DESTDIR)$(datadir)/wireplumber/wireplumber.conf.d audio/reform-hdmi-audio-priority.conf
 	$(INSTALL)     -d $(DESTDIR)$(sysconfdir)/profile.d
