diff --git a/drivers/gpu/drm/display/drm_bridge_connector.c b/drivers/gpu/drm/display/drm_bridge_connector.c
index a2d30cf9e06d..48a3c12b3a01 100644
--- a/drivers/gpu/drm/display/drm_bridge_connector.c
+++ b/drivers/gpu/drm/display/drm_bridge_connector.c
@@ -214,8 +214,6 @@ drm_bridge_connector_detect(struct drm_connector *connector, bool force)
 
 		if (hdmi)
 			drm_atomic_helper_connector_hdmi_hotplug(connector, status);
-
-		drm_bridge_connector_hpd_notify(connector, status);
 	} else {
 		switch (connector->connector_type) {
 		case DRM_MODE_CONNECTOR_DPI:
@@ -230,6 +228,8 @@ drm_bridge_connector_detect(struct drm_connector *connector, bool force)
 		}
 	}
 
+	drm_bridge_connector_hpd_notify(connector, status);
+
 	return status;
 }
 
