commit 60ac0396b40b19c8d4c8015a25226fe414ab576b
Author: Lukas F. Hartmann <lukas@mntre.com>
Date:   Thu Aug 8 20:51:57 2024 +0200

    rk3588: dsi2: fix dw_mipi_dsi2_remove signature

diff --git a/drivers/gpu/drm/rockchip/dw-mipi-dsi2-rockchip.c b/drivers/gpu/drm/rockchip/dw-mipi-dsi2-rockchip.c
index e978f95..8e87527 100644
--- a/drivers/gpu/drm/rockchip/dw-mipi-dsi2-rockchip.c
+++ b/drivers/gpu/drm/rockchip/dw-mipi-dsi2-rockchip.c
@@ -1441,12 +1441,14 @@ static int dw_mipi_dsi2_probe(struct platform_device *pdev)
 	return ret;
 }
 
-static void dw_mipi_dsi2_remove(struct platform_device *pdev)
+static int dw_mipi_dsi2_remove(struct platform_device *pdev)
 {
 	struct dw_mipi_dsi2 *dsi2 = platform_get_drvdata(pdev);
 
 	component_del(&pdev->dev, &dw_mipi_dsi2_ops);
 	mipi_dsi_host_unregister(&dsi2->host);
+
+	return 0;
 }
 
 static __maybe_unused int dw_mipi_dsi2_runtime_suspend(struct device *dev)
