--- a/arch/arm64/boot/dts/rockchip/rk3588-firefly-icore-3588q.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-firefly-icore-3588q.dtsi
@@ -45,39 +45,48 @@
 	cpu-supply = <&vdd_cpu_lit_s0>;
 };
 
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi0>, <&hdptxphy_hdmi1>;
+	clock-names = "hdmi0_phy_pll", "hdmi1_phy_pll";
+};
+
 &i2c0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c0m2_xfer>;
 	status = "okay";
 
-	vdd_cpu_big0_s0: regulator@42 {
+	vdd_cpu_big0_s0: vdd_cpu_big0_mem_s0: rk8602@42 {
 		compatible = "rockchip,rk8602";
 		reg = <0x42>;
 		fcs,suspend-voltage-selector = <1>;
 		regulator-always-on;
 		regulator-boot-on;
+		regulator-compatible = "rk860x-reg";
 		regulator-min-microvolt = <550000>;
 		regulator-max-microvolt = <1050000>;
 		regulator-name = "vdd_cpu_big0_s0";
 		regulator-ramp-delay = <2300>;
 		vin-supply = <&vcc5v0_sys>;
+		rockchip,suspend-voltage-selector = <1>;
 
 		regulator-state-mem {
 			regulator-off-in-suspend;
 		};
 	};
 
-	vdd_cpu_big1_s0: regulator@43 {
-		compatible = "rockchip,rk8603", "rockchip,rk8602";
+	vdd_cpu_big1_s0: vdd_cpu_big1_mem_s0: rk8603@43 {
+		compatible = "rockchip,rk8602", "rockchip,rk8603";
 		reg = <0x43>;
 		fcs,suspend-voltage-selector = <1>;
 		regulator-always-on;
 		regulator-boot-on;
+		regulator-compatible = "rk860x-reg";
 		regulator-min-microvolt = <550000>;
 		regulator-max-microvolt = <1050000>;
 		regulator-name = "vdd_cpu_big1_s0";
 		regulator-ramp-delay = <2300>;
 		vin-supply = <&vcc5v0_sys>;
+		rockchip,suspend-voltage-selector = <1>;
 
 		regulator-state-mem {
 			regulator-off-in-suspend;
@@ -127,18 +136,19 @@
 	pinctrl-0 = <&spi2m2_cs0 &spi2m2_pins>;
 	status = "okay";
 
-	pmic@0 {
+	rk806single: pmic@0 {
 		compatible = "rockchip,rk806";
 		reg = <0x0>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
 		gpio-controller;
 		#gpio-cells = <2>;
-		pinctrl-names = "default";
+		pinctrl-names = "default", "pmic-power-off";
 		pinctrl-0 = <&pmic_pins>, <&rk806_dvs1_null>,
 			    <&rk806_dvs2_null>, <&rk806_dvs3_null>;
+		pinctrl-1 = <&rk806_dvs1_pwrdn>;
+		pmic-reset-func = <1>;
 		spi-max-frequency = <1000000>;
-		system-power-controller;
 
 		vcc1-supply = <&vcc5v0_sys>;
 		vcc2-supply = <&vcc5v0_sys>;
@@ -157,22 +167,23 @@
 		vcca-supply = <&vcc5v0_sys>;
 
 		rk806_dvs1_null: dvs1-null-pins {
-			pins = "gpio_pwrctrl1";
+			pins = "gpio_pwrctrl2";
 			function = "pin_fun0";
 		};
 
-		rk806_dvs2_null: dvs2-null-pins {
+		rk806_dvs2_null: rk806_dvs2_null {
 			pins = "gpio_pwrctrl2";
 			function = "pin_fun0";
 		};
 
-		rk806_dvs3_null: dvs3-null-pins {
+		rk806_dvs3_null: rk806_dvs3_null {
 			pins = "gpio_pwrctrl3";
 			function = "pin_fun0";
 		};
 
 		regulators {
 			vdd_gpu_s0: vdd_gpu_mem_s0: dcdc-reg1 {
+				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <550000>;
 				regulator-max-microvolt = <950000>;
@@ -212,11 +223,12 @@
 				};
 			};
 
-			vdd_vdenc_s0: vdd_vdenc_mem_s0: dcdc-reg4 {
+			vdd_vdenc_s0: dcdc-reg4 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <550000>;
 				regulator-max-microvolt = <950000>;
+				regulator-init-microvolt = <750000>;
 				regulator-ramp-delay = <12500>;
 				regulator-name = "vdd_vdenc_s0";
 
@@ -434,7 +446,74 @@
 				};
 			};
 		};
+
+		rk806_dvs1_pwrdn: rk806_dvs1_pwrdn {
+			function = "pin_fun2";
+			phandle = <0xc0>;
+			pins = "gpio_pwrctrl1";
+		};
+
+		rk806_dvs1_rst: rk806_dvs1_rst {
+			function = "pin_fun3";
+			pins = "gpio_pwrctrl1";
+		};
+
+		rk806_dvs1_slp: rk806_dvs1_slp {
+			function = "pin_fun1";
+			pins = "gpio_pwrctrl1";
+		};
+
+		rk806_dvs2_dvs: rk806_dvs2_dvs {
+			function = "pin_fun4";
+			pins = "gpio_pwrctrl2";
+		};
+
+		rk806_dvs2_gpio: rk806_dvs2_gpio {
+			function = "pin_fun5";
+			pins = "gpio_pwrctrl2";
+		};
+
+		rk806_dvs2_pwrdn: rk806_dvs2_pwrdn {
+			function = "pin_fun2";
+			pins = "gpio_pwrctrl2";
+		};
+
+		rk806_dvs2_rst: rk806_dvs2_rst {
+			function = "pin_fun3";
+			pins = "gpio_pwrctrl2";
+		};
+
+		rk806_dvs2_slp: rk806_dvs2_slp {
+			function = "pin_fun1";
+			pins = "gpio_pwrctrl2";
+		};
+
+		rk806_dvs3_dvs: rk806_dvs3_dvs {
+			function = "pin_fun4";
+			pins = "gpio_pwrctrl3";
+		};
+
+		rk806_dvs3_gpio: rk806_dvs3_gpio {
+			function = "pin_fun5";
+			pins = "gpio_pwrctrl3";
+		};
+
+		rk806_dvs3_pwrdn: rk806_dvs3_pwrdn {
+			function = "pin_fun2";
+			pins = "gpio_pwrctrl3";
+		};
+
+		rk806_dvs3_rst: rk806_dvs3_rst {
+			function = "pin_fun3";
+			pins = "gpio_pwrctrl3";
+		};
+
+		rk806_dvs3_slp: rk806_dvs3_slp {
+			function = "pin_fun1";
+			pins = "gpio_pwrctrl3";
+		};
 	};
+
 };
 
 &uart2 {
--- a/arch/arm64/boot/dts/rockchip/rk3588-mnt-reform2.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-mnt-reform2.dts
@@ -16,14 +16,33 @@
 
 / {
 	model = "MNT Reform 2 with RCORE RK3588 Module";
-	compatible = "mntre,reform2-rcore", "firefly,icore-3588q", "rockchip,rk3588";
-	chassis-type = "laptop";
+	compatible = "mntre,reform2-rcore", "firefly,icore3588q", "rockchip,rk3588";
 
 	aliases {
 		ethernet0 = &gmac0;
 		mmc1 = &sdmmc;
 	};
 
+	analog-sound {
+		compatible = "audio-graph-card";
+		dais = <&i2s0_8ch_p0>;
+		label = "rk3588-wm8960";
+		routing =
+			"Headphone Jack", "HP_L",
+			"Headphone Jack", "HP_R",
+			"Ext Spk", "SPK_LP",
+			"Ext Spk", "SPK_LN",
+			"Ext Spk", "SPK_RP",
+			"Ext Spk", "SPK_RN",
+			"LINPUT1", "Mic Jack",
+			"Mic Jack", "MICB";
+		widgets =
+			"Headphone", "Headphone Jack",
+			"Microphone", "Mic Jack",
+			"Speaker", "Ext Spk";
+
+	};
+
 	chosen {
 		stdout-path = "serial2:1500000n8";
 	};
@@ -43,7 +62,7 @@
 		clock-output-names = "gmac0_clkin";
 	};
 
-	pcie30_avdd1v8: regulator-pcie30-avdd1v8 {
+	pcie30_avdd1v8: pcie30-avdd1v8 {
 		compatible = "regulator-fixed";
 		regulator-boot-on;
 		regulator-always-on;
@@ -53,7 +72,7 @@
 		vin-supply = <&avcc_1v8_s0>;
 	};
 
-	pcie30_avdd0v75: regulator-pcie30-avdd0v75 {
+	pcie30_avdd0v75: pcie30-avdd0v75 {
 		compatible = "regulator-fixed";
 		regulator-always-on;
 		regulator-boot-on;
@@ -63,7 +82,7 @@
 		vin-supply = <&avdd_0v75_s0>;
 	};
 
-	vcc12v_dcin: regulator-vcc12v-dcin {
+	vcc12v_dcin: vcc12v-dcin-regulator {
 		compatible = "regulator-fixed";
 		regulator-always-on;
 		regulator-boot-on;
@@ -72,7 +91,7 @@
 		regulator-name = "vcc12v_dcin";
 	};
 
-	vcc_1v1_nldo_s3: regulator-vcc-1v1-nldo-s3 {
+	vcc_1v1_nldo_s3: vcc-1v1-nldo-s3 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_1v1_nldo_s3";
 		regulator-always-on;
@@ -82,7 +101,7 @@
 		vin-supply = <&vcc5v0_sys>;
 	};
 
-	vcc3v3_pcie30: regulator-vcc3v3-pcie30 {
+	vcc3v3_pcie30: vcc3v3-pcie30 {
 		compatible = "regulator-fixed";
 		regulator-always-on;
 		regulator-boot-on;
@@ -92,7 +111,7 @@
 		vin-supply = <&vcc12v_dcin>;
 	};
 
-	vcc5v0_host: regulator-vcc5v0-host {
+	vcc5v0_host: vcc5v0-host {
 		compatible = "regulator-fixed";
 		regulator-always-on;
 		regulator-boot-on;
@@ -101,7 +120,7 @@
 		regulator-name = "vcc5v0_host";
 	};
 
-	vcc5v0_sys: regulator-vcc5v0-sys {
+	vcc5v0_sys: vcc5v0-sys {
 		compatible = "regulator-fixed";
 		regulator-always-on;
 		regulator-boot-on;
@@ -111,7 +130,7 @@
 		vin-supply = <&vcc12v_dcin>;
 	};
 
-	vcc5v0_usb: regulator-vcc5v0-usb {
+	vcc5v0_usb: vcc5v0-usb {
 		compatible = "regulator-fixed";
 		regulator-always-on;
 		regulator-boot-on;
@@ -126,6 +145,54 @@
 	status = "okay";
 };
 
+&combphy1_ps {
+	status = "okay";
+};
+
+&combphy2_psu {
+	status = "okay";
+};
+
+&cpu_b0 {
+	cpu-supply = <&vdd_cpu_big0_s0>;
+	mem-supply = <&vdd_cpu_big0_s0>;
+};
+
+&cpu_b1 {
+	cpu-supply = <&vdd_cpu_big0_s0>;
+	mem-supply = <&vdd_cpu_big0_s0>;
+};
+
+&cpu_b2 {
+	cpu-supply = <&vdd_cpu_big1_s0>;
+	mem-supply = <&vdd_cpu_big1_s0>;
+};
+
+&cpu_b3 {
+	cpu-supply = <&vdd_cpu_big1_s0>;
+	mem-supply = <&vdd_cpu_big1_s0>;
+};
+
+&cpu_l0 {
+	cpu-supply = <&vdd_cpu_lit_s0>;
+	mem-supply = <&vdd_cpu_lit_mem_s0>;
+};
+
+&cpu_l1 {
+	cpu-supply = <&vdd_cpu_lit_s0>;
+	mem-supply = <&vdd_cpu_lit_mem_s0>;
+};
+
+&cpu_l2 {
+	cpu-supply = <&vdd_cpu_lit_s0>;
+	mem-supply = <&vdd_cpu_lit_mem_s0>;
+};
+
+&cpu_l3 {
+	cpu-supply = <&vdd_cpu_lit_s0>;
+	mem-supply = <&vdd_cpu_lit_mem_s0>;
+};
+
 &gmac0 {
 	clock_in_out = "output";
 	phy-handle = <&rgmii_phy>;
@@ -161,17 +228,77 @@
 	status = "okay";
 };
 
+&hdmi0 {
+	enable-gpios = <&gpio4 RK_PA0 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
+&hdmi0_in {
+	hdmi0_in_vp2: endpoint {
+		remote-endpoint = <&vp2_out_hdmi0>;
+	};
+};
+
+&hdmi1 {
+	enable-gpios = <&gpio4 RK_PB4 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi1>;
+	};
+};
+
+&hdptxphy_hdmi1 {
+	status = "okay";
+};
+
 &i2c6 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c6m0_xfer>;
 	status = "okay";
 
+	wm8960: codec@1a {
+		compatible = "wlf,wm8960";
+		reg = <0x1a>;
+		clocks = <&cru I2S0_8CH_MCLKOUT>;
+		clock-names = "mclk";
+		assigned-clocks = <&cru I2S0_8CH_MCLKOUT>;
+		assigned-clock-rates = <12288000>;
+		#sound-dai-cells = <0>;
+
+		port {
+			wm8960_p0: endpoint {
+				remote-endpoint = <&i2s0_8ch_p0_0>;
+			};
+		};
+	};
+
 	rtc@68 {
 		compatible = "nxp,pcf8523";
 		reg = <0x68>;
 	};
 };
 
+&i2s0_8ch {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s0_lrck
+		&i2s0_mclk
+		&i2s0_sclk
+		&i2s0_sdi0
+		&i2s0_sdo0>;
+	status = "okay";
+
+	i2s0_8ch_p0: port {
+		i2s0_8ch_p0_0: endpoint {
+			dai-format = "i2s";
+			mclk-fs = <256>;
+			remote-endpoint = <&wm8960_p0>;
+		};
+	};
+};
+
 &mdio0 {
 	rgmii_phy: ethernet-phy@0 {
 		compatible = "ethernet-phy-ieee802.3-c22";
@@ -224,6 +351,10 @@
 	};
 };
 
+&i2c3 {
+	status = "okay";
+};
+
 &pwm8 {
 	pinctrl-0 = <&pwm8m2_pins>;
 	status = "okay";
@@ -248,6 +379,21 @@
 	status = "okay";
 };
 
+&spi1 {
+	status = "okay";
+	assigned-clocks = <&cru CLK_SPI1>;
+	assigned-clock-rates = <200000000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi1m2_cs0 &spi1m2_pins>;
+	num-cs = <1>;
+
+	spidev@0 {
+		compatible = "mntre,lpc11u24";
+		spi-max-frequency = <1000000>;
+		reg = <0>;
+	};
+};
+
 &tsadc {
 	status = "okay";
 };
@@ -328,6 +474,20 @@
 	status = "okay";
 };
 
+&vp0 {
+	vp0_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp0>;
+	};
+};
+
+&vp2 {
+	vp2_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
+		remote-endpoint = <&hdmi0_in_vp2>;
+	};
+};
+
 &vp2 {
 	vp2_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
 		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
