commit 9dad5f62224a7c51dfbc2cfaaa11e11ed6b55e3a
Author: Lucie L. Hartmann <lukas@mntre.com>
Date:   Thu May 21 16:18:34 2026 +0200

    qcs8550/adreno:: double the HFI message timeout

diff --git a/drivers/gpu/drm/msm/adreno/a6xx_hfi.c b/drivers/gpu/drm/msm/adreno/a6xx_hfi.c
index 53cfdf4..706ac3b 100644
--- a/drivers/gpu/drm/msm/adreno/a6xx_hfi.c
+++ b/drivers/gpu/drm/msm/adreno/a6xx_hfi.c
@@ -111,7 +111,7 @@ static int a6xx_hfi_wait_for_msg_interrupt(struct a6xx_gmu *gmu, u32 id, u32 seq
 	do {
 		/* Wait for a response */
 		ret = gmu_poll_timeout(gmu, REG_A6XX_GMU_GMU2HOST_INTR_INFO, val,
-			val & A6XX_GMU_GMU2HOST_INTR_INFO_MSGQ, 100, 1000000);
+			val & A6XX_GMU_GMU2HOST_INTR_INFO_MSGQ, 100, 2000000);
 
 		if (!ret)
 			break;
