commit af5c226a9178cc211b6c7a0f3e969f7e32df78f6
Author: Lucie L. Hartmann <lukas@mntre.com>
Date:   Thu May 21 16:41:33 2026 +0200

    qcs8550/adreno: debug: lower devfreq threshold to 40

diff --git a/drivers/gpu/drm/msm/msm_gpu_devfreq.c b/drivers/gpu/drm/msm/msm_gpu_devfreq.c
index 2e1d5c3..7cb18d5 100644
--- a/drivers/gpu/drm/msm/msm_gpu_devfreq.c
+++ b/drivers/gpu/drm/msm/msm_gpu_devfreq.c
@@ -152,7 +152,7 @@ void msm_devfreq_init(struct msm_gpu *gpu)
 	 * where due to stalling waiting for vblank we could get stuck
 	 * at (for ex) 30fps at 50% utilization.
 	 */
-	priv->gpu_devfreq_config.upthreshold = 50;
+	priv->gpu_devfreq_config.upthreshold = 40;
 	priv->gpu_devfreq_config.downdifferential = 10;
 
 	mutex_init(&df->lock);
