diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index d5cdd77e5..853e95d80 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -854,7 +854,26 @@ malidp0: display@f080000 {

 			port {
 				dpi0_out: endpoint {
+					remote-endpoint = <&dp1_out>;
+				};
+			};
+		};

+		hdptx0: display@f200000 {
+			compatible = "cdn,ls1028a-dp";
+			reg = <0x0 0xf200000 0x0 0xfffff>;
+			interrupts = <0 221 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clockgen QORIQ_CLK_HWACCEL 2>, <&clockgen QORIQ_CLK_HWACCEL 2>,
+			         <&clockgen QORIQ_CLK_HWACCEL 2>, <&clockgen QORIQ_CLK_HWACCEL 2>,
+			         <&clockgen QORIQ_CLK_HWACCEL 2>, <&dpclk>;
+			clock-names = "clk_core", "pclk", "sclk",
+			              "cclk", "clk_vif", "clk_pxl";
+			lane-mapping = <0x4e>;
+			edp_num_lanes = <0x2>;
+
+			port {
+				dp1_out: endpoint {
+					remote-endpoint = <&dpi0_out>;
 				};
 			};
 		};
