diff --git a/src/view/xdg-shell.cpp b/src/view/xdg-shell.cpp
index 4e3c898e..43a630a7 100644
--- a/src/view/xdg-shell.cpp
+++ b/src/view/xdg-shell.cpp
@@ -107,6 +107,10 @@ std::shared_ptr<wayfire_xdg_popup> wayfire_xdg_popup::create(wlr_xdg_popup *popu
 
     self->surface_root_node = std::make_shared<wayfire_xdg_popup_node>(self->get_id());
     self->set_surface_root_node(self->surface_root_node);
+
+    // initially move the popup out of the way of the pointer
+    self->surface_root_node->set_offset({INT_MAX, INT_MAX});
+
     self->set_output(self->popup_parent->get_output());
     self->unconstrain();
     return self;
