diff --git a/debian/bin/gencontrol.py b/debian/bin/gencontrol.py
index 02d457dd3..e49c2f585 100755
--- a/debian/bin/gencontrol.py
+++ b/debian/bin/gencontrol.py
@@ -74,13 +74,9 @@ class Gencontrol(Base):
         for env, attr, desc in self.env_flags:
             setattr(self, attr, False)
             if os.getenv(env):
-                if self.changelog[0].distribution == 'UNRELEASED':
-                    import warnings
-                    warnings.warn(f'Disable {desc} on request ({env} set)')
-                    setattr(self, attr, True)
-                else:
-                    raise RuntimeError(
-                        f'Unable to disable {desc} in release build ({env} set)')
+                import warnings
+                warnings.warn(f'Disable {desc} on request ({env} set)')
+                setattr(self, attr, True)
 
     def _setup_makeflags(self, names, makeflags, data):
         for src, dst, optional in names:
diff --git a/debian/config/arm64/defines b/debian/config/arm64/defines
index e64418af6..486876d23 100644
--- a/debian/config/arm64/defines
+++ b/debian/config/arm64/defines
@@ -3,11 +3,10 @@ cross-compile-compat: arm-linux-gnueabihf-
 kernel-arch: arm64
 featuresets:
  none
- rt
 
 [build]
 image-file: arch/arm64/boot/Image
-signed-code: true
+signed-code: false
 vdso: true
 
 [image]
diff --git a/debian/config/arm64/none/defines b/debian/config/arm64/none/defines
index b754cf65e..93d1fe516 100644
--- a/debian/config/arm64/none/defines
+++ b/debian/config/arm64/none/defines
@@ -1,7 +1,6 @@
 [base]
 flavours:
  arm64
- cloud-arm64
 default-flavour: arm64
 quick-flavour: cloud-arm64
 
diff --git a/debian/config/defines b/debian/config/defines
index 113926e5f..5acb3fa1f 100644
--- a/debian/config/defines
+++ b/debian/config/defines
@@ -171,3 +171,7 @@ gcc-12: gcc-12 <!stage1 !cross !pkg.linux.nokernel>, gcc-12-@gnu-type-package@ <
 # initramfs-generators
 initramfs-fallback: linux-initramfs-tool
 initramfs-tools: initramfs-tools (>= 0.120+deb8u2)
+
+[packages]:
+installer: false
+docs: false
diff -Nru linux-6.4.11/debian/rules.real linux-6.4.11/debian/rules.real
--- linux-6.4.11/debian/rules.real	2023-08-17 09:05:43.000000000 +0200
+++ linux-6.4.11/debian/rules.real	2023-09-01 06:43:41.000000000 +0200
@@ -43,7 +43,10 @@
 setup_env += DISTRIBUTION_OFFICIAL_BUILD=1 DISTRIBUTOR="$(DISTRIBUTOR)" DISTRIBUTION_VERSION="$(SOURCEVERSION)" KBUILD_BUILD_TIMESTAMP="$(SOURCE_DATE)" KBUILD_BUILD_VERSION_TIMESTAMP="$(DISTRIBUTOR) $(SOURCEVERSION) ($(SOURCE_DATE_UTC_ISO))" KBUILD_BUILD_USER="$(word 1,$(subst @, ,$(MAINTAINER)))" KBUILD_BUILD_HOST="$(word 2,$(subst @, ,$(MAINTAINER)))"
 setup_env += KBUILD_VERBOSE=$(if $(filter terse,$(DEB_BUILD_OPTIONS)),0,1)
 
-MAKE_CLEAN = $(setup_env) $(MAKE) KCFLAGS=-fdebug-prefix-map=$(CURDIR)/= KBUILD_HOSTCFLAGS='$(CFLAGS) $(CPPFLAGS)' HOSTCFLAGS='$(CFLAGS) $(CPPFLAGS)' KBUILD_HOSTLDFLAGS='$(LDFLAGS)'
+CFLAGS_FOR_BUILD   ?= $(shell dpkg-architecture --host-arch $(DEB_BUILD_ARCH) --force --command dpkg-buildflags --get CFLAGS)
+CPPFLAGS_FOR_BUILD ?= $(shell dpkg-architecture --host-arch $(DEB_BUILD_ARCH) --force --command dpkg-buildflags --get CPPFLAGS)
+LDFLAGS_FOR_BUILD  ?= $(shell dpkg-architecture --host-arch $(DEB_BUILD_ARCH) --force --command dpkg-buildflags --get LDFLAGS)
+MAKE_CLEAN = $(setup_env) $(MAKE) KCFLAGS=-fdebug-prefix-map=$(CURDIR)/= KBUILD_HOSTCFLAGS='' HOSTCFLAGS='' KBUILD_HOSTLDFLAGS=''
 MAKE_SELF := $(MAKE) -f debian/rules.real $(MAKEOVERRIDES)
 MAKEOVERRIDES =
 
