--- a/debian/lib/python/debian_linux/debian.py
+++ b/debian/lib/python/debian_linux/debian.py
@@ -192,6 +192,7 @@
     )
 )
 (?:\+b\d+)?
+(?:\+reform[0-9]+T[0-9]+Z[0-9]*)?
 $
     """, re.X)
 
diff --git a/debian/bin/gencontrol.py b/debian/bin/gencontrol.py
index 02d457dd3..e49c2f585 100755
--- a/debian/bin/gencontrol.py
+++ b/debian/bin/gencontrol.py
@@ -74,13 +74,9 @@ class Gencontrol(Base):
         for env, attr, desc in self.env_flags:
             setattr(self, attr, False)
             if os.getenv(env):
-                if self.changelog[0].distribution == 'UNRELEASED':
-                    import warnings
-                    warnings.warn(f'Disable {desc} on request ({env} set)')
-                    setattr(self, attr, True)
-                else:
-                    raise RuntimeError(
-                        f'Unable to disable {desc} in release build ({env} set)')
+                import warnings
+                warnings.warn(f'Disable {desc} on request ({env} set)')
+                setattr(self, attr, True)
 
     def _setup_makeflags(self, names, makeflags, data):
         for src, dst, optional in names:
