diff --git a/debian/bin/gencontrol.py b/debian/bin/gencontrol.py
index 02d457dd3..e49c2f585 100755
--- a/debian/bin/gencontrol.py
+++ b/debian/bin/gencontrol.py
@@ -74,13 +74,9 @@ class Gencontrol(Base):
         for env, attr, desc in self.env_flags:
             setattr(self, attr, False)
             if os.getenv(env):
-                if self.changelog[0].distribution == 'UNRELEASED':
-                    import warnings
-                    warnings.warn(f'Disable {desc} on request ({env} set)')
-                    setattr(self, attr, True)
-                else:
-                    raise RuntimeError(
-                        f'Unable to disable {desc} in release build ({env} set)')
+                import warnings
+                warnings.warn(f'Disable {desc} on request ({env} set)')
+                setattr(self, attr, True)
 
     def _setup_makeflags(self, names, makeflags, data):
         for src, dst, optional in names:
diff --git a/debian/config/arm64/defines b/debian/config/arm64/defines
index e64418af6..486876d23 100644
--- a/debian/config/arm64/defines
+++ b/debian/config/arm64/defines
@@ -3,11 +3,10 @@ cross-compile-compat: arm-linux-gnueabihf-
 kernel-arch: arm64
 featuresets:
  none
- rt
 
 [build]
 image-file: arch/arm64/boot/Image
-signed-code: true
+signed-code: false
 vdso: true
 
 [image]
diff --git a/debian/config/arm64/none/defines b/debian/config/arm64/none/defines
index b754cf65e..93d1fe516 100644
--- a/debian/config/arm64/none/defines
+++ b/debian/config/arm64/none/defines
@@ -1,7 +1,6 @@
 [base]
 flavours:
  arm64
- cloud-arm64
 default-flavour: arm64
 quick-flavour: cloud-arm64
 
diff --git a/debian/config/defines b/debian/config/defines
index 113926e5f..5acb3fa1f 100644
--- a/debian/config/defines
+++ b/debian/config/defines
@@ -171,3 +171,7 @@ gcc-12: gcc-12 <!stage1 !cross !pkg.linux.nokernel>, gcc-12-@gnu-type-package@ <
 # initramfs-generators
 initramfs-fallback: linux-initramfs-tool
 initramfs-tools: initramfs-tools (>= 0.120+deb8u2)
+
+[packages]:
+installer: false
+docs: false
diff --git a/debian/rules.real b/debian/rules.real
index 9bb0c617b..daf5b388e 100644
--- a/debian/rules.real
+++ b/debian/rules.real
@@ -485,8 +485,8 @@ binary_udeb binary_udeb_test: export KW_CONFIG_DIR=debian/installer
 binary_udeb:
 	dh_testdir
 	dh_prep
-	kernel-wedge install-files $(ABINAME)
-	kernel-wedge check $(PACKAGE_NAMES)
+	#kernel-wedge install-files $(ABINAME)
+	#kernel-wedge check $(PACKAGE_NAMES)
 	dh_fixperms
 	dh_gencontrol
 	dh_builddeb -- $(BUILDDEB_ARGS)
